update readme links (#361)
tinybirdco/use-case-examples: iterate_snowflake/README.md

- Create a new git branch 
- Create a new Snowflake .datasource data file in the branch, as in the example PR. You shouldn't need fixtures as the Data Source will be populated from the connection, but you can add fixtures and/or tests if needed.
  - For information about Snowflake options read the [connector documentation](https://www.tinybird.co/docs/ingest/snowflake)
- Commit your code and create a new PR/MR from the branch to `main` 
- Wait for CI to finish.

> - Check that everything is working as expected in the temporary branch, CI will automatically create it and name it like `tmp_ci_{PRid}`).

Reviewer: Could we be more specific? How do I check that everything is working, do I need to `sync now` the data source?

- Merge and wait for CD. Now you should have your Data Source in the main workspace, using the connection, and the temporary CI branch will be deleted.

[Example Pull Request](https://github.com/tinybirdco/use-case-examples/pull/359)

### Update Snowflake Data Source with a new column, with same connection

- Create another git branch
- Add the new column to the schema and query
- Commit your code and create an ew PR from the branch to `main`
- Wait for CI to finish.

> - Check that everything is working as expected in the temporary branch, CI will automatically create it and name it like `tmp_ci_{PRid}`).

Reviewer: Same here. How do we check that it works? Does the connector works when it was previously created?

Author: Same that above

Reviewer: are we 100% sure about this? I thought connectors didn't work in branches (except if you create them for the first time), so I'd expect the connector not to work if you modify the data source, not sure if I'm missing anything 🙏

Author: I've made a ton of tests, but going to make another 😅

Author: I've left a couple of tests over https://app.tinybird.co/gcp/europe-west3/asun_20240909/datasource/created_no_fixture
* When you create a tb branch directly (in this case `from_ui_connection_dont_work` the connection doesn't works (surprise!), idem if you create the branch with cli, etc
* But when you run CI (branch `tmp_ci__5`) the connection is OK, I understand this happens b/c not only the branch is created, but then the changes in the datasource are applied.
In fact it's the same we're telling the users to do, if you create a branch the connector breaks and you must re-create the ds. I've tested that in the branch `from_cli_to_push`:
* I've create the branch from main and checked that the DS didn't work
* Then I've manually `tb push -f` the data source and now it works

Reviewer: Perfect then, thanks!

- Merge and wait for CD. Now you should have your Data Source updated in the main workspace, using the connection, and the temporary CI branch will be deleted.

[Example Pull Request](https://github.com/tinybirdco/use-case-examples/pull/360)

### Delete a Snowflake Data Source

- Create another git branch
- Remove the .datasource data file

> - Commit your code and create an ew PR from the branch to `main`

Reviewer: ```suggestion
- Commit your code and create a new PR from the branch to `main`
```

Same typo in other places 🙏

- Wait for CI to finish.
- Check that everything is working as expected in the temporary branch, CI will automatically create it and name it like `tmp_ci_{PRid}`).

> - Merge and wait for CD. Now you should have your Data Source updated in the main workspace, using the connection, and the temporary CI branch will be deleted.

Reviewer: `Now you should have your Data Source updated in the main workspace`

The DS is deleted right? (only if it did not have dependencies I guess)

What about the connection is it still there?

Author: Added a note about dependencies an about the connection.
